framework/filters/PageCache.php

```diff
@@ -123,6 +124,13 @@ class PageCache extends ActionFilter
      * @since 2.0.4
      */
     public $cacheHeaders = true;
+    /**
+     * @var array a list of placeholders for embedding dynamic contents. This property
+     * is used internally to implement the content caching feature. Do not modify it.
+     * @internal
+     * @since 2.0.11
+     */
```

Comment:
Me too. But it's how [it works](#r92295170)…

I can improve it later with a getter/setter, some `DynamicContentInterface` with this stuff (`getDynamicPlacholders`, `setDynamicPlacholders`), a type hinting in appropriate `View` methods and a some documentation for other developers.
Of course if it's welcome. ;)